src/java.base/share/classes/javax/crypto/KeyAgreement.java

```diff
@@ -659,9 +659,16 @@ public final int generateSecret(byte[] sharedSecret, int offset)
      * {@code generateSecret} to change the private information used in
      * subsequent operations.
      *
-     * @param algorithm the requested secret-key algorithm
-     *
-     * @return the shared secret key
+     * @param algorithm the requested secret-key algorithm. This is
```

Comment:
Did you consider instead referencing the new section you are adding for standard key algorithm names (https://bugs.openjdk.org/browse/JDK-8346736)? It could be reworded as:

"See the Key Algorithms section in the Java Security Standard Algorithm Names Specification for information about standard secret key algorithm names. Use "Generic" ..."

Reply:
Good idea.

Reply:
Do you suggest me adding the same to `encapsulate` and `decapsulate` of `KEM` in this PR?

Reply:
Yes.

Reply:
Finally, no. KEM changes will be in https://github.com/openjdk/jdk/pull/22929.